File: 241211/정수 명령 처리/process-numeric-commands.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int n, num;
stack<int> st;
int main() {
    cin >> n;
    for(int i=0; i<n; i++){
        string s;
        cin >> s;
        if(s == "push"){
            cin >> num;
            st.push(num);
        }
        else if(s == "size"){
            cout << st.size() <<'\n';
        }
        else if(s == "empty"){
            cout << st.empty() <<'\n';
        }
        else if(s == "pop"){
            cout << st.top() <<'\n';
            st.pop();
        }
        else if(s == "top"){
            cout << st.top() <<'\n';
        }
    }
    return 0;
}

File: 250221/정수 명령 처리/process-numeric-commands.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();
        int n = sc.nextInt();

        for(int i=0; i<n; i++){
            String s = sc.next();

            if(s.equals("push")){
                int num = sc.nextInt();
                stack.push(num);
            }
            else if(s.equals("top")) System.out.println(stack.peek());
            else if(s.equals("size")) System.out.println(stack.size());
            else if(s.equals("empty")){
                if(stack.empty()) System.out.println(1);
                else System.out.println(0);
            }
            else System.out.println(stack.pop());
        }
        sc.close();
    }
}